DuckDuckGo/HomeViewController+HomeScreenDelegate.swift

```diff
@@ -22,11 +22,23 @@ import EasyTipView
 
 extension HomeViewController: HomeScreenTipsDelegate {
     
-    func showPrivateSearchTip() {
+    func showPrivateSearchTip(didShow: @escaping (Bool) -> Void) {
 
         DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
-            guard let superView = self?.parent?.view else { return }
-            guard let self = self else { return }
+            guard let self = self else {
+                didShow(false)
+                return
+            }
+            
+            guard !self.omniBarTextFieldHasFocus else {
+                didShow(false)
+                return
+            }
+            
+            guard let superView = self.parent?.view else {
+                didShow(false)
+                return
+            }
```

Comment:
What do you think about combining these guards into a single one? This can be also applied to other files from this PR.

Reply:
Of course 🤦‍♂️